source/features/pr-filters.tsx

```diff
@@ -60,43 +62,63 @@ function addDraftFilter({delegateTarget: reviewsFilter}: DelegateEvent): void {
 	addDropdownItem(dropdown, 'Not ready for review (Draft PR)', 'draft', 'true');
 }
 
-async function addStatusFilter(): Promise<void> {
+const cachedChecksStatus = cache.function(async () => {
+	const {repository} = await api.v4(`repository(${getRepoGQL()}) {
+		head: object(expression: "HEAD") {
+			... on Commit {
+				history(first: 10) {
+					nodes {
+						status {
+							state
+						}
+					}
+				}
+			}
+		}
+	}`);
+
+	return repository.head.history.nodes.some((commit: AnyObject) => commit.status);
+}, {
+	expiration: 3,
+	cacheKey: () => __featureName__ + ':' + getRepoURL()
+});
+
+async function addChecksFilter(): Promise<void> {
 	const reviewsFilter = select(reviewsFilterSelector);
 	if (!reviewsFilter) {
 		return;
 	}
 
-	// TODO: replace this with an API call
-	const hasCI = await fetchCIStatus();
+	const hasCI = await cachedChecksStatus();
 	if (!hasCI) {
 		return;
 	}
 
 	// Copy existing element and adapt its content
-	const statusFilter = reviewsFilter.cloneNode(true);
-	statusFilter.id = '';
+	const checksFilter = reviewsFilter.cloneNode(true);
+	checksFilter.id = '';
 
-	select('summary', statusFilter)!.firstChild!.textContent = 'Status\u00A0'; // Only replace text node, keep caret
-	select('.SelectMenu-title', statusFilter)!.textContent = 'Filter by build status';
+	select('summary', checksFilter)!.firstChild!.textContent = 'Checks\u00A0'; // Only replace text node, keep caret
+	select('.SelectMenu-title', checksFilter)!.textContent = 'Filter by checks status';
 
-	const dropdown = select('.SelectMenu-list', statusFilter)!;
+	const dropdown = select('.SelectMenu-list', checksFilter)!;
 	dropdown.textContent = ''; // Drop previous filters
 
 	for (const status of ['Success', 'Failure', 'Pending']) {
 		addDropdownItem(dropdown, status, 'status', status.toLowerCase());
 	}
 
-	reviewsFilter.after(' ', statusFilter);
+	reviewsFilter.after(checksFilter);
 }
 
 function init(): void {
 	delegate(reviewsFilterSelector, 'toggle', addDraftFilter, true);
-	addStatusFilter();
+	addChecksFilter();
 }
 
 features.add({
 	id: __featureName__,
-	description: 'Adds `Build status` and draft PR dropdown filters in PR lists.',
+	description: 'Adds Checks and draft PR dropdown filters in PR lists.',
```

Comment:
```suggestion
	description: 'Adds Checks and Draft PR dropdown filters in PR lists.',
```

The readme and screenshot should also be updated (arrow not necessary, as long as the Checks dropdown is open)